fil_fungible_token/src/receiver/types.rs

```diff
@@ -15,9 +16,12 @@ pub trait FrcXXXTokenReceiver {
     fn token_received(params: TokenReceivedParams);
 }
 
-#[derive(Serialize_tuple, Deserialize_tuple)]
+#[derive(Serialize_tuple, Deserialize_tuple, PartialEq, Eq, Clone, Debug)]
 pub struct TokenReceivedParams {
-    pub sender: Address,
+    /// Address of the sender or operator that initiated the transfer
+    pub sender: ActorID,
+    /// The account that the tokens are being pulled from (the token actor address itself for mint)
+    pub from: ActorID,
```

Comment:
It's an interesting question whether these values should be ActorId or Address. Certainly, if Address, it should be an ID type address. And so passing ID here makes that very clear. But everywhere else (e.g. syscalls, built-in actors) always pass addresses. Perhaps for FIP discussion.

Reply:
A new thought: I think we should add the `to` here as well. Yes the receiver will know their own address, but including it explicitly will allow the message to be transparently forwarded, used with proxy patterns etc. ERC-777 includes it (where delegate calls etc make the need more obvious). FVM in the future might add delegation capabilites that make this relevant.